### DISCUSSION TITLE
Nuxt new project

### DISCUSSION THREAD

**Discussion by ghost** [nuxt/nuxt#19036] [Questions] [answered]
Hello guys, do I need these files in my nuxt project?  I mean dev, components.d.ts and etc? They make my files hard to read and reach, in Nuxt 2 project was free from this much trash in main directory so, if they are neccessary can I make it more organized or smth?
![image](https://user-images.githubusercontent.com/103945399/218831717-e94c0531-dcd4-4379-a6ec-5e58259f8a14.png)

**Comment by AloisSeckar** [comment, marked as answer]
`.nuxt` directory only contains files from local dev build, I don't think you should worry about them too much. This is not your final distribution.

`.d.ts` files describe TypeScript types, so they are somewhat important (although after half a year delved in TS development I haven't figured out yet wheter they are **required** to make TS project work or just nice to have for making dev's life easier).

But definitely you can benefit from them during development. At least in VS Code, while they help to navigate from usages to declarations on Ctril+LMB. Since Nuxt does a lot of auto-imports, it is not always clear where the method/component comes from. But they are linked through these definition files. Active Volar extension is required to make it work though.

**Reply by ghost** [reply]
Thank you! So I should keep all of them, right?

**Reply by AloisSeckar** [reply]
Yes. Keep them there and let Nuxt doing its job. It will recreate them anyway when you next launch the dev server :)
